### PR TITLE
Homework 15

### PR DESCRIPTION
**Pull request by olark1d**
Add homework 15 and wait for a feedback:)

### CODE REVIEW

File: HOMEWORK_15/EX1.CPP [hunk 1]
```diff
@@ -0,0 +1,240 @@
+#include <iostream>
+#include <algorithm>
+#include <string>
+#include <vector>
+
+struct Student
+{
+    std::string name;
+    int marks[4];
```
Comment (bohdanpc) [inline HOMEWORK_15/EX1.CPP]: replace `4` with const variable and use it instead everywhere

File: HOMEWORK_15/EX1.CPP [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+{
+    std::sort(students, students + size, compare);
+}
+
+Student* mostSuccessfulStudents(Student* students, int size)
+{
+    Student* mostSuccessful = &students[0];
+    for(int i = 1; i < size; i++)
+    {
+        if(students[i].averageMark > mostSuccessful->averageMark)
```
Comment (bohdanpc) [inline HOMEWORK_15/EX1.CPP:53]: if `averageMark` function was never called for a particular student then the `averageMark` field will be unset. That's is quite a big assumption you make here that someone called it

File: HOMEWORK_15/EX1.CPP (hunk 1 above)
Comment (olark1d) [reply HOMEWORK_15/EX1.CPP]: Done

File: HOMEWORK_15/EX1.CPP (hunk 2 above)
Comment (olark1d) [reply HOMEWORK_15/EX1.CPP:53]: Done

File: HOMEWORK_15/EX1.CPP [hunk 3]
```diff
@@ -3,44 +3,44 @@
 #include <string>
 #include <vector>
 
+#define Marks 4
```
Comment (bohdanpc) [inline HOMEWORK_15/EX1.CPP:6]: constexpr unsigned MarksCount = 4;

**Pull request closed without merging**